app/models/MessageId.scala

```diff
@@ -17,34 +17,24 @@
 package models
 
 import play.api.mvc.PathBindable
+import play.api.libs.json.Json
+import play.api.libs.json.Format
 
-final class MessageId private (val index: Int) {
-  override def toString: String = s"MessageId($index)"
-
-  override def equals(obj: Any): Boolean = obj match {
-    case x: MessageId => x.index == this.index
-    case _            => false
-  }
-
-  def publicValue: Int = index + 1
-
-}
+case class MessageId(index: Int)
```

Comment:
Should we rename index here since this is no longer the index but instead the messageId value. As suggested above it may make sense to have an index property so we don't make off by one errors anywhere.

So the interface for this should be something like:
```
trait {
  def value: Int // value that will come in from an request
  def index: Int // corresponds to the position in the db, which is only needed for compatibility with some services that use position
}
```

Reply:
Ah you beat me to it haha 😆